File: Release/WetiMatlabFunctions/DataFunctions/CalculatePitchTravel.m
function PitchTravel= CalculatePitchTravel(Data,Time,StartTime)
% Calculates pitch travel from a pitch angle signal.
% Function for CalculateStatistics.

% inputs
arguments
    Data                (:,1) double
    Time                (:,1) double
    StartTime           (1,1) double
end

PitchRate   = gradient(Data,Time);
dt          = Time(2)-Time(1); % assuming equally spaced time
PitchTravel = sum(abs(PitchRate(Time>=StartTime)))*dt;

end

File: Release/PythonFunctions/DataFunctions/CalculatePitchTravel.py
import numpy as np

def CalculatePitchTravel(data, time, start_time):
    pitch_rate = np.gradient(data,time)

    dt = time[1] - time[0]

    pitch_travel = np.sum(np.abs(pitch_rate[time >= start_time])) * dt

    return pitch_travel
